bravado/fido_client.py

```diff
@@ -148,4 +149,8 @@ def result(self, timeout=None):
             return self._eventual_result.wait(timeout=timeout)
         except crochet.TimeoutError:
             self._eventual_result.cancel()
-            raise
+            raise fido.exceptions.HTTPTimeoutError(
```

Comment:
I think it's fine - the original traceback doesn't really give any more info as to what's going on. It looks like this:

    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "...lib/python3.6/site-packages/bravado/http_future.py", line 70, in result
        inner_response = self.future.result(timeout=timeout)
      File "...lib/python3.6/site-packages/bravado/fido_client.py", line 147, in result
        return self._eventual_result.wait(timeout=timeout)
      File "...lib/python3.6/site-packages/crochet/_eventloop.py", line 245, in wait
        result = self._result(timeout)
      File "...lib/python3.6/site-packages/crochet/_eventloop.py", line 209, in _result
        raise TimeoutError()
    crochet._eventloop.TimeoutError

I think having a consistent exception type is far more important - the traceback only really matters when debugging internal issues (i.e. when developing bravado / fido).